tests/Skills/Parent/SkillsHelper.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core.Skills;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class SkillsHelper
+    {
+        private readonly BotFrameworkSkill _botFrameworkSkill;
+        private readonly SkillConversationIdFactoryBase _conversationIdFactory;
+        private readonly string _botId;
+        private readonly SkillHttpClient _skillHttpClient;
+        private readonly BotAdapter _botAdapter;
+        private readonly IExtendedUserTokenProvider _tokenExchangeProvider;
+        private readonly string _parentConnectionName;
+        private readonly Uri _callbackUri;
+
+        public SkillsHelper(IConfiguration configuration, SkillHttpClient skillHttpClient, BotAdapter botAdapter, SkillConversationIdFactoryBase conversationIdFactory)
+        {
+            // We use a single skill in this example.
+            var section = configuration.GetSection("BotFrameworkSkill");
+            _botFrameworkSkill = section?.Get<BotFrameworkSkill>();
+            _botId = configuration.GetSection(MicrosoftAppCredentials.MicrosoftAppIdKey)?.Value;
+            _skillHttpClient = skillHttpClient;
+            _botAdapter = botAdapter;
+            _tokenExchangeProvider = botAdapter as IExtendedUserTokenProvider;
+            _parentConnectionName = configuration.GetSection("ConnectionName")?.Value;
+            _callbackUri = new Uri(configuration.GetSection("CallbackUri")?.Value);
+            _conversationIdFactory = conversationIdFactory;
+        }
+
+        public async Task<InvokeResponse<ExpectedReplies>> PostActivityAsync(Activity activity, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var response = await _skillHttpClient.PostActivityAsync<ExpectedReplies>(_botId, _botFrameworkSkill, _callbackUri, activity, cancellationToken);
+            return response;
+        }
+
+        public async Task<bool> InterceptOAuthCards(ITurnContext turnContext, Activity[] activities, CancellationToken cancellationToken)
+        {
+            var activity = activities.FirstOrDefault(a => a.Type == ActivityTypes.Message);
+            if (activity != null)
+            {
+                return await InterceptOAuthCards(turnContext, activity, cancellationToken);
+            }
+
+            return false;
+        }
+
+        public async Task<bool> InterceptOAuthCards(ITurnContext turnContext, Activity activity, CancellationToken cancellationToken)
```

Comment:
Once it's cleaned up a bit, this nasty little dance seems like it could be moved into the SDK to avoid having developers write this code.

Regardless, I think a comment or two in this method would help maintainers and consumers of samples a lot.